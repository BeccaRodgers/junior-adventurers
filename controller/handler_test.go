package controller

import (
	"bytes"
	"context"
	approvals "github.com/approvals/go-approval-tests"
	"github.com/stretchr/testify/assert"
	"github.com/stretchr/testify/require"
	"github.com/yosssi/gohtml"
	"junior-adventurers/fixtures"
	"junior-adventurers/memdb"
	"net/http"
	"net/http/httptest"
	"testing"
)

func TestHandler_Homepage(t *testing.T) {
	approveGetPage(t, "/")
}

func TestHandler_GuildPage(t *testing.T) {
	approveGetPage(t, "/guilds/1")
}

func TestHandler_NewMemberPage(t *testing.T) {
	approveGetPage(t, "/members")
}

func approveGetPage(t *testing.T, path string) {
	t.Helper()

	guilds := memdb.NewGuildRepository()
	require.NoError(t, fixtures.InsertGuilds(context.TODO(), guilds))
	members := memdb.NewMemberRepository()
	require.NoError(t, fixtures.InsertMembers(context.TODO(), members))

	handler := Handler(guilds, members)
	request := httptest.NewRequest(http.MethodGet, path, nil)
	recorder := httptest.NewRecorder()
	handler.ServeHTTP(recorder, request)
	assert.Equal(t, http.StatusOK, recorder.Code)
	bodyBytes := gohtml.FormatBytes(recorder.Body.Bytes())
	var formattedOutput bytes.Buffer
	formattedOutput.Write(bodyBytes)
	approvals.Verify(t, &formattedOutput)
}
