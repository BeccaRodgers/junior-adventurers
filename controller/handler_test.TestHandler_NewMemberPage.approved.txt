<!doctype html>
<html lang="en">
  <head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>
      Junior Adventurers
    </title>
    <link rel="stylesheet" href="/static/style.min.css">
    <link rel="icon" type="image/x-icon" href="/static/favicon.ico">
    <script src="/static/htmx.min.js"></script>
  </head>
  <body hx-target="this">
    <section class="section">
      <div class="container has-text-centered">
        <h1 class="title">
          New Member Details
        </h1>
      </div>
      <div class="container">
        <div class="field">
          <label class="label">
            Name
          </label>
          <div class="control">
            <input class="input" type="text" placeholder="Tiffany Aching">
          </div>
        </div>
        <div class="field">
          <label class="label">
            Date of Birth
          </label>
          <div class="control">
            <input class="input" type="date">
          </div>
        </div>
        <div class="field">
          <label class="label">
            Species
          </label>
          <div class="control">
            <div class="select">
              <select>
                <option>
                  Select Species
                </option>
                <option>
                  Human
                </option>
                <option>
                  Dwarf
                </option>
                <option>
                  Werewolf
                </option>
              </select>
            </div>
          </div>
        </div>
        <div class="field">
          <label class="label">
            Guild
          </label>
          <div class="control">
            <div class="select">
              <select>
                <option>
                  Select Guild
                </option>
                <option>
                  Founder's Guild
                </option>
              </select>
            </div>
          </div>
        </div>
        <div class="field is-grouped">
          <p class="control">
            <button class="button is-link" hx-get="/" hx-push-url="true">
              Home
            </button>
          </p>
          <p class="control">
            <button class="button is-primary">
              Submit
            </button>
          </p>
        </div>
      </div>
    </section>
  </body>
</html>